### ISSUE TITLE
Filtering vcf with expression

### ISSUE THREAD

**Issue by rabiafidan** [samtools/bcftools#1604]
Hi,
I want to filter positions from a single-sample mpileup vcf (not called). I want to get positions where any of alternative alleles has AD>5 and the fraction of reads supporting that allele is above a threshold. For that, I figured firstly I would apply these filters on the first alt allele and then the second alt allele and so on then merge the results. When I apply the filters on the first alt allele, it works fine.
```
$ bcftools view -i 'INFO/DP > 10 & INFO/AD[1] > 5 & (INFO/AD[1])/(INFO/DP) > 0.02' sample.vcf | \
> bcftools query -f '%POS\t%DP\t%AD\n' | head
837	2858	1115,314,0
913	1430	0,1429,0
1043	1117	859,257,0
1244	1148	840,308,0
1488	66	9,24,0
1603	2345	2231,112,1,0
1806	4626	2046,267,0
1843	2314	1953,357,2,0
1897	2482	2278,200,2,0
1971	174	139,34
```
 But when I apply the filters on the second alt allele, the result is as follows:
 ```
$ bcftools view -i 'INFO/DP > 10 & INFO/AD[2] > 5 & (INFO/AD[2])/(INFO/DP) > 0.02' sample.vcf| \
>bcftools query -f '%POS\t%DP\t%AD\n' | head
323	367	367,0
324	367	367,0
325	369	369,0
326	370	370,0
327	370	370,0
328	370	370,0
330	370	370,0
331	370	370,0
332	370	370,0
333	370	370,0
```
Is this the expected output?   Thank you!

```
$ bcftools --version
bcftools 1.13
Using htslib 1.13
Copyright (C) 2021 Genome Research Ltd.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

**Comment by pd3** [comment]
Thanks for reporting the problem, this is now fixed by #45923484.

Note that one can also do this:
```
bcftools +fill-tags input.vcf -- -t VAF | bcftools view -i 'VAF>0.02'
```

**Comment by rabiafidan** [comment]
Oh great, thanks!
